Modify demo_helmholtz.py to solve for plane wave solution in complex mode

jzitelli opened the pull request:
This is a minimal change addressing the issue raised in https://github.com/FEniCS/dolfinx/issues/3557

garth-wells commented:
We aim to have demos and tests run (near) identical code in real and complex cases. Otherwise testing is problematic. Can you make the real case run the same code? Setting the NumPy type should discard the complex part when computing with reals.

Also, run `ruff` to format the Python code.

jzitelli commented:
Thank you for the feedback, I was indeed able to consolidate the real/complex code paths.  In the real mode the script will simply solve for the real part of the complex exponential (I updated the introductory comments accordingly).  I also fixed the lint errors found using ruff.

garth-wells commented:
Resolves #3557
